### PR TITLE
🐛Destination-clickhouse: enabled and fixed tests for normalization

### LINKED ISSUE #12582: :bug: Destination Clickhouse : Normalization doesn't work in the acceptance tests.
## Current Behavior
The acceptance tests don't check any cases with normalization due to some problems with port sharing inside the container.
All tests with normalization are disabled. Also, we run DAT tests without normalization due to the false value in the `supportsNormalization` method.

Screenshot from 2022-05-04 22-26-36

## Expected Behavior
Run all test cases using normalization.

### PR DESCRIPTION
**Pull request by etsybaev**
## What
The acceptance tests don't check any cases with normalization due to some problems with port sharing inside the container.
All tests with normalization are disabled. Also, we run DAT tests without normalization due to the false value in the supportsNormalization method.

## How
Enabled tests for normalization and fixed all issues found by those (Python part fixes have been already merged as part of https://github.com/airbytehq/airbyte/pull/13591 PR).

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No impact, only tests have been updated in this PR

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py [hunk 1]
```diff
@@ -549,7 +549,7 @@ def cast_property_type(self, property_name: str, column_name: str, jinja_column:
                 return f"try_parse({replace_operation} as {sql_type}) as {column_name}"
             if self.destination_type == DestinationType.CLICKHOUSE:
                 sql_type = jinja_call("type_date()")
-                return f"parseDateTimeBestEffortOrNull(trim(BOTH '\"' from {replace_operation})) as {column_name}"
+                return f"toDate(parseDateTimeBestEffortOrNull(trim(BOTH '\"' from {replace_operation}))) as {column_name}"
```
Comment (bazarnov) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:552]: Was returning `toDate(...)` helpful to convert `date_time` back to `date` in your case?

Comment (bazarnov) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:552]: Do you have local tests passed?

Comment (etsybaev) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:552]: @bazarnov yes, it helped.

**Comment by bazarnov** [comment]
What are next steps with changes to `normalization` ? Do we need to create follow-up issue?

**Comment by etsybaev** [comment]
> What are next steps with changes to `normalization` ? Do we need to create follow-up issue?

@bazarnov not sure. I just need to have it merged and released but have no idea what is the workflow. Can I publish and merge it along with my java changes?

**Comment by bazarnov** [comment]
You can after all tests are successful. Have you checked the failed test results?
________________________________

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java [hunk 2]
```diff
@@ -6,13 +6,34 @@
...
 import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ClickhouseTestDataComparator extends AdvancedTestDataComparator {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(ClickhouseTestDataComparator.class);
   private final ExtendedNameTransformer namingResolver = new ExtendedNameTransformer();
 
+  public static final String CLICKHOUSE_DATETIME_WITH_TZ_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSX";
```
Comment (mkhokh-33) [inline airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java:25]: seems this constant is used only inside of this class, could we change it to private?

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import io.airbyte.db.jdbc.JdbcSourceOperations;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class ClickhouseTestSourceOperations extends JdbcSourceOperations {
+
+  public static final DateTimeFormatter CLICKHOUSE_DATETIME_WITH_TZ_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSSX");
```
Comment (mkhokh-33) [inline airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java:18]: do we use this constant?

**Review by mkhokh-33 — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java (hunk 3 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java:18]: Yup, it's used in parseDestinationDateWithTz and compareDateTimeValues methods

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java]: Goog catch. Yup, not it's used in this class only. Updated, thanks

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestinationAcceptanceTest.java [hunk 4]
```diff
@@ -84,9 +86,15 @@ protected JsonNode getConfig() {
     // Note: ClickHouse official JDBC driver uses HTTP protocol, its default port is 8123
     // dbt clickhouse adapter uses native protocol, its default port is 9000
     // Since we disabled normalization and dbt test, we only use the JDBC port here.
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseDestinationAcceptanceTest.java]: good catch, updated. Thanks

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestDataComparator.java]: Goog catch. Yup, not it's used in this class only. Updated, thanks

File: airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java (hunk 3 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-clickhouse/src/test-integration/java/io/airbyte/integrations/destination/clickhouse/ClickhouseTestSourceOperations.java]: You were right, It didn't. Updated, thanks

**Review by bazarnov — APPROVED** [review]
LGTM, but please, update the brach to the latest master and sync up with @annalvova05 to avoid `normalization` version conflicts.

**Review by mkhokh-33 — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]
LGTM. we can proceed with merge without airbyte review

**Comment by etsybaev** [comment]
Strict-encrypt part will be updated as part of https://app.zenhub.com/workspaces/java-connectors-625f0102b11304001e0f6c6e/issues/airbytehq/airbyte/14858

**Pull request merged**